src/shared/contexts/editor/index.tsx

```diff
@@ -15,16 +15,30 @@ import {
 } from 'src/shared/utils/fluxFunctions'
 import {getFluxExample} from 'src/shared/utils/fluxExample'
 
+// LSP
+import {
+  ExecuteCommand,
+  ExecuteCommandArgument,
+} from 'src/languageSupport/languages/flux/lsp/utils'
+import LspConnectionManager from 'src/languageSupport/languages/flux/lsp/connection'
+
 // Utils
 import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 import {CLOUD} from 'src/shared/constants'
 
 export interface EditorContextType {
   editor: EditorType | null
-  setEditor: (editor: EditorType) => void
+  setEditor: (
+    editor: EditorType,
+    conn: React.MutableRefObject<LspConnectionManager>
+  ) => void
   inject: (options: InjectionOptions) => void
   injectFunction: (fn, cbToParent) => void
   injectVariable: (variableName, cbToParent) => void
+  injectViaLsp: (
+    cmd: ExecuteCommand,
+    data: Omit<ExecuteCommandArgument, 'textDocument'>
+  ) => void
```

Comment:
This is the method which would be used, to inject the schema.

```
const {injectViaLsp} = useContext(EditorContext)

// look at the type definitions for ExecuteCommandArgument, for the payload
injectViaLsp(ExecuteCommand.injectMeasurement, payload) 
```

Note that the LSP injection requires that the bucket, measurement, field, and value EACH be injected one at a time...in series.